Update to 20.04

ElectrodeYT opened the pull request.

walking-octopus commented:
Hi. Thanks for your contribution, I really didn't have the time to get around to it lately, and it's nice to finally have one place to start working on ironing out the issues with the Focal transition. I saw your message on Telegram, and have copied the relevant information from there.

As I said, I've started the review to outline some of my concerns. I remember there were a few issues the last time @muhammad23012009 tried porting it to Focal, so it would be nice to iron them out before merging. When completed, I think this should be merged into a separate `focal` branch that would become the main one. Changes would first be done on `focal`, and then backported to `xenial` until some time after it's officially deprecated and most device ports have made the transition.

I once again thank you for helping out with this, and hope this can be completed and merged as soon as can be. Thanks for helping out!

ElectrodeYT commented:
No problem! Personally, I would create a new branch called `xenial` and rename `main` to `focal`.

The distro detection would likely require a PyPi package. `platform.linux_distribution()` was deprecated and removed, replaced with the `distro` package.

I honestly don't know whats wrong with the Waydroid OTA version detection. Could be that i have not updated it for a while?

walking-octopus commented:
> No problem! Personally, I would create a new branch called `xenial` and rename `main` to `focal`.

That's what I was planning to do. I'd get around to it when we're done with ironing out the Focal port's bugs.

> The distro detection would likely require a PyPi package. `platform.linux_distribution()` was deprecated and removed, replaced with the `distro` package.

Using PyPi packages with Clickable is just too much of a pain. I think a small RegExp to get PRETTY_NAME out of /etc/os-release would be easier.

> I honestly don't know whats wrong with the Waydroid OTA version detection. Could be that I have not updated it for a while?

I remember @muhammad23012009 had reported similar issue in his Focal porting attempt. We'd have to see...

ElectrodeYT commented:
That should be both of the first bugs done - the Waydroid issue was that the response seems to be given as a string (converted it to int first), and the os-release file is now being passed with the CSV library (this is a bit of a hack, but using a RegEx seems like a bit more of a hack to me personally)

eheintzmann commented:
Hi,

I looked at the changes made by @ElectrodeYT in this PR, and they all seem correct to me (most of them are just replacements of “Ubuntu” with “Lomiri” or “16.04” with “20.04”).

So I built the click package (with GitHub Actions, see : https://github.com/eheintzmann/ElectrodeYT_system-info-ut/actions/runs/10545151333)  and installed it on my UT Xiaomi POCO X3 NFC to test this PR.

Everything seems to **work perfectly**, all the fields are filled with correct information.
In particular, 
- Ubuntu version (distro) is “Ubuntu 20.04.6 LTS”
- Waydroid/Lineage Version is correct (but there is a “.zip” at the end of the string)

Let me know if I can do other tests or provide screenshots or help in any other way…
